pkg/controller/pod_driver.go

```diff
@@ -86,19 +163,167 @@ func (p *PodDriver) podDeletedHandler(ctx context.Context, pod *corev1.Pod) (rec
 		// do nothing
 		return reconcile.Result{}, nil
 	}
-
+	if util.IsPodResourceError(pod) {
+		klog.V(5).Infof("the pod is PodResourceError, podDeletedHandler skip delete the pod:%s", pod.Name)
+		return reconcile.Result{}, nil
+	}
 	// todo
-
 	klog.V(5).Infof("Remove finalizer of pod %s namespace %s", pod.Name, pod.Namespace)
 	controllerutil.RemoveFinalizer(pod, config.Finalizer)
 	if err := p.Client.Update(ctx, pod); err != nil {
+		klog.Errorf("Update pod err:%v", err)
 		return reconcile.Result{}, err
 	}
+	// do recovery
+	klog.V(6).Infof("Annotations:%v", pod.Annotations)
+	if pod.Annotations == nil {
+		return reconcile.Result{}, nil
+	}
+	var targets = make([]string, 0)
+	for k, v := range pod.Annotations {
+		if k == util.GetReferenceKey(v) {
+			targets = append(targets, v)
+			// umount target
+			//for {
+			//	//klog.V(5).Infof("umount target:%s\n", v)
+			//	mntErr := p.Mounter.Unmount(v) // may mount repeated
+			//	if mntErr != nil {
+			//		klog.V(5).Infof("umount get err:%v", mntErr)
+			//		break
+			//	}
+			//}
+		}
+	}
+	if len(targets) == 0 {
+		return reconcile.Result{}, nil
+	}
+	// create the pod even if get err
+	defer func() {
+		// check pod delete
+		key := types.NamespacedName{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+		}
+		// check pod exists
+		for i := 0; i < 30; i++ {
+			if err := p.Client.Get(ctx, key, pod); err == nil {
+				klog.Infof("pod still exists, wait to create")
+				time.Sleep(time.Second * 5)
+			} else {
+				if apierrors.IsNotFound(err) {
+					break
+				}
+				klog.Errorf("get pod err:%v", err) // create pod even if get err
+				break
+			}
+
+		}
+		var newPod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        pod.Name,
+				Namespace:   pod.Namespace,
+				Labels:      pod.Labels,
+				Annotations: pod.Annotations,
+			},
+			Spec: pod.Spec,
+		}
+		controllerutil.AddFinalizer(newPod, config.Finalizer)
+		err := p.Client.Create(ctx, newPod)
+		if err != nil {
+			klog.Errorf("create pod:%s err:%v", pod.Name, err)
+		}
+	}()
+	cmd := pod.Spec.Containers[0].Command
+	if cmd == nil || len(cmd) < 3 {
+		klog.Errorf("get error pod command:%v", cmd)
+		return reconcile.Result{}, nil
+	}
+	sourcePath, _, err := util.ParseMntPath(cmd[2])
+	if err != nil {
+		klog.Error(err)
+		return reconcile.Result{}, nil
+	}
+
+	klog.Infof("start umount :%s", sourcePath)
+	if err := p.Mounter.Unmount(sourcePath); err != nil {
```

Comment:
If mountPoint in node is umounted manually,  mount pod will be error and then be deleted by manager, but umount function here will be error.